driver-core/src/main/com/mongodb/internal/connection/netty/NettyStream.java

```diff
@@ -260,7 +252,13 @@ public void writeAsync(final List<ByteBuf> buffers, final AsyncCompletionHandler
             composite.addComponent(true, ((NettyByteBuf) cur).asByteBuf().retain());
         }
 
+        long writeTimeoutMS = operationContext.getTimeoutContext().getWriteTimeoutMS();
+        Optional<WriteTimeoutHandler> writeTimeoutHandler = writeTimeoutMS != NO_SCHEDULE_TIME
+                ? Optional.of(new WriteTimeoutHandler(writeTimeoutMS, MILLISECONDS)) : Optional.empty();
+        writeTimeoutHandler.map(w -> channel.pipeline().addBefore("ChannelInboundHandlerAdapter", "WriteTimeoutHandler", w));
```

Comment:
Thank you for pointing this out. The `writeHandler` used in the lambda expression needs to be final or effectively final, which complicates using a simple if statement due to variable mutability. That’s why `Optional` was utilized here. I understand this can be confusing, so I've refactored this part to enhance readability. Please let me know if you have any suggestions on how to simplify it further without using `Optional`.